# libcss: utils
# 3 files, 60 changed lines
#
# Regenerate: tools/regen-patches.sh && tools/split-patches.py
#
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libcss/src/utils/parserutilserror.h browser/libcss/src/utils/parserutilserror.h
--- upstream/libcss/src/utils/parserutilserror.h	2026-09-20 23:27:15.843751866 +0100
+++ browser/libcss/src/utils/parserutilserror.h	2026-09-20 23:19:17.259710250 +0100
@@ -18,12 +18,8 @@
  * \param error  The ParserUtils error to convert
  * \return The corresponding LibCSS error
  */
-static inline css_error css_error_from_parserutils_error(
-		parserutils_error error)
-{
-	/* Currently, there's a 1:1 mapping, so we've nothing to do */
-	return (css_error) error;
-}
+/* MacSurf: out-of-line. Defined in macos9_extra_stubs.c. */
+extern css_error css_error_from_parserutils_error(parserutils_error error);
 
 #endif
 
diff -urN upstream/libcss/src/utils/utils.c browser/libcss/src/utils/utils.c
--- upstream/libcss/src/utils/utils.c
+++ browser/libcss/src/utils/utils.c
@@ -5,7 +5,17 @@
  * Copyright 2007-9 John-Mark Bell <jmb@netsurf-browser.org>
  */
 
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <libwapcaplet/libwapcaplet.h>
+#include <libcss/errors.h>
+#include <libcss/fpmath.h>
+
 #include "utils/utils.h"
+
+css_fixed css__number_from_string(const uint8_t *data, size_t len,
+		bool int_only, size_t *consumed);
 
 css_fixed css__number_from_lwc_string(lwc_string *string,
 		bool int_only, size_t *consumed)
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libcss/src/utils/utils.h browser/libcss/src/utils/utils.h
--- upstream/libcss/src/utils/utils.h	2026-09-20 23:27:15.843751866 +0100
+++ browser/libcss/src/utils/utils.h	2026-09-20 23:19:17.259851389 +0100
@@ -39,42 +39,10 @@
 css_fixed css__number_from_string(const uint8_t *data, size_t len,
 		bool int_only, size_t *consumed);
 
-static inline bool isDigit(uint8_t c)
-{
-	return '0' <= c && c <= '9';
-}
-
-static inline bool isHex(uint8_t c)
-{
-	return isDigit(c) || ('a' <= c && c <= 'f') || ('A' <= c && c <= 'F');
-}
-
-static inline uint32_t charToHex(uint8_t c)
-{
-	c -= '0';
-
-	if (c > 9)
-		c -= 'A' - '9' - 1;
-
-	if (c > 15)
-		c -= 'a' - 'A';
-
-	return c;
-}
-
-static inline css_error css_error_from_lwc_error(lwc_error err)
-{
-        switch (err) {
-        case lwc_error_ok:
-                return CSS_OK;
-        case lwc_error_oom:
-                return CSS_NOMEM;
-        case lwc_error_range:
-                return CSS_BADPARM;
-        default:
-                break;
-        }
-        return CSS_INVALID;
-}
+/* MacSurf: out-of-line. Defined in macos9_extra_stubs.c. */
+extern bool isDigit(uint8_t c);
+extern bool isHex(uint8_t c);
+extern uint32_t charToHex(uint8_t c);
+extern css_error css_error_from_lwc_error(lwc_error err);
 
 #endif
